resources/shared/test-runner.mjs

```diff
@@ -41,7 +52,7 @@ export class TestRunner {
             }
             performance.mark(syncStartLabel);
             const syncStartTime = performance.now();
-            this.#test.run(this.#page);
+            await this._runSyncStep(this.test, this.page);
```

Comment:
This seems to have a side effect of always pushing the measurement end until when a promise resolves.
I don't think we want to do that for rAF based measurement method (since that can affect the score)
so maybe we should check the return value of this function and await conditionally?

Reply:
More concretely, the following code logs: 1, 2, 3:
```
(async function() { console.log(1); await (() => { })(); console.log(3); })(); console.log(2);
```
whereas the following code logs 1, 3, 2:
```
(async function() { console.log(1); (() => { })(); console.log(3); })(); console.log(2);
```

i.e. `await` will always delay the execution of the continuation until the end of the current micro-task. While that's highly desirable for async measurement, I don't think we want to affect rAF measurement.